catroid/src/org/catrobat/catroid/content/Look.java

```diff
@@ -303,12 +311,47 @@ public float getDirectionInUserInterfaceDimensionUnit() {
 		return direction;
 	}
 
+	public float getRealDirectionInUserInterfaceDimensionUnit() {
+		return realRotation;
+	}
+
+	public void setRotationMode(int mode) {
+		rotationMode = mode;
+	}
+
 	public void setDirectionInUserInterfaceDimensionUnit(float degrees) {
-		setRotation((-degrees + DEGREE_UI_OFFSET) % 360);
+		float oldRotation = getDirectionInUserInterfaceDimensionUnit();
+		rotation = (-degrees + DEGREE_UI_OFFSET) % 360;
+		realRotation = degrees;
+		float newRotation = getDirectionInUserInterfaceDimensionUnit();
+
+		switch (rotationMode) {
+			case ROTATION_STYLE_LEFT_RIGHT_ONLY:
+				setRotation(0f);
+				if ((oldRotation < 0 && newRotation >= 0) || (oldRotation >= 0 && newRotation < 0)) {
```

Comment:
Can you use a boolean for this statement. It should improve the readability of the statement.